shark/shark/src/main/java/shark/AndroidReferenceReaderFactory.kt

```diff
@@ -17,9 +17,19 @@ class AndroidReferenceReaderFactory(
       listOf(
         JavaLocalReferenceReader(graph, referenceMatchers),
       ) +
-        OpenJdkInstanceRefReaders.values().mapNotNull { it.create(graph) } +
-        ApacheHarmonyInstanceRefReaders.values().mapNotNull { it.create(graph) } +
-        AndroidReferenceReaders.values().mapNotNull { it.create(graph) }
+        AndroidReferenceReaders.values().mapNotNull { it.create(graph) } +
+        (
+          OpenJdkInstanceRefReaders.values().mapNotNull { it.create(graph) } +
+            ApacheHarmonyInstanceRefReaders.values().mapNotNull { it.create(graph) }
+          )
```

Comment:
This code would strongly benefit from some intermediate variables... so much happening in this one statement

Reply:
This was a dumb hack to get this going. I think the decision to wrap with FlatteningPartitionedStructReferenceReader should be made for each custom ref reader, rather than applied or all nothing to an enum that provides a bunch of ref readers.